src/tara_ros.cpp

```diff
@@ -149,377 +227,736 @@ taraCamera::taraCamera(ros::NodeHandle _comm_nh, ros::NodeHandle _param_nh) :
+			return ;
+		}
+		else
+		{
+			//Reading the configuration to verify the values are set 
+			uStatus = GetIMUConfig(&lIMUConfig);
+			if(!uStatus)			
+			{
+				cout << "GetIMUConfig Failed\n";			
+				return ;
 			}
-			foutRight << "]\nprojection_matrix:\n   rows: 3\n   cols: 4\n   data: [";
-		}
-		if ( !strcmp (buffer, "P1"))
-   		{
-   			for (int iterator = 0; iterator < 12; iterator ++)
-   			{
-				foutLeft << extrinsic_matrix["P1"]["data"][iterator];
-				if (iterator < 12 - 1)
+			//Finding the sampling interval time
+			glIMU_Interval = GetIMUIntervalTime(lIMUConfig);
+		}
+
+		//Configuring IMU update mode
+		lIMUInput.IMU_UPDATE_MODE = IMU_CONT_UPDT_EN;
+		lIMUInput.IMU_NUM_OF_VALUES = IMU_AXES_VALUES_MIN;
+
+		//Setting the IMU update mode using HID command
+		uStatus = ControlIMUCapture(&lIMUInput);
+		if(uStatus == FALSE)			
+		{
+			cout << "ControlIMUCapture Failed\n";
+			return ;
+		}
+	}
+
+	void taraCamera::IMU_enable()
+	{
+		cout << endl  << "		IMU Sample Application " << endl  << endl;
+		cout << " Application to illustrate the IMU unit LSM6DS0 integrated with Tara Camera" << endl;
+		cout << " Demonstrating the rotations of camera around x-axis and y-axis " << endl;
+		cout << " IMU values are limited from -90 to +90 degrees for illustration " << endl << endl;
+
+		SetIMUConfigDefaultEnable();
+
+		//Allocating buffers for output structure			
+		lIMUOutput = (IMUDATAOUTPUT_TypeDef*)malloc(1 * sizeof(IMUDATAOUTPUT_TypeDef));
+
+		//Memory validation
+		if(lIMUOutput == NULL)
+		{
+			cout << "Memory Allocation for output failed\n";
+			return ;
+		}
+
+		lIMUOutput->IMU_VALUE_ID = 0;
+
+		if (GetIMUValueBuffer_write() == false )
+		{
+			cout <<"GetIMUValueBuffer_write failed" << endl;
+		}
+
+		while (ros::ok() )
+		{
+			//HID command
+			if( !GetIMUValueBuffer ( lIMUOutput ) )
+			{
+				cout << "GetIMUValueBuffer failed\n";
+			}
+			//Calculating angles based on the current raw values from IMU			
+			getInclination(lIMUOutput->gyroX, lIMUOutput->gyroY, lIMUOutput->gyroZ, lIMUOutput->accX, lIMUOutput->accY, lIMUOutput->accZ);
+		}
+	}
+
+	int taraCamera::econ_strcmp (const char *str1,const char *str2)
+	{
+		int iter = 0;
+		for (iter = 0; str1 [iter] && str2 [iter] ; iter ++ )
+		{
+			if (str1[iter] != str2[iter] )
+			{
+				if ( str1 [iter] > str2 [iter] )
 				{
-					foutLeft << ", ";
+					return 1;
+				}
+				else
+				{
+					return -1;
 				}
 			}
-			foutLeft << "]\n";
-		}
-		if ( !strcmp (buffer, "P2"))
-   		{
-			for (int iterator = 0; iterator < 12; iterator ++)
-   			{
-				foutRight << extrinsic_matrix["P2"]["data"][iterator];
-				if (iterator < 12 - 1)
+		}
+
+		if ( str1 [iter] > str2 [iter] )
+		{
+			return 1;
+		}
+		else
+		{
+			if ( str1 [iter] < str2 [iter] )
+			{
+				return -1;
+			}
+			else
+			{
+				return 0;
+			}
+		}
+	}
+	
+	BOOL taraCamera::checkFirmware (UINT8 MajorVersion, UINT8 MinorVersion1, UINT16 MinorVersion2, UINT16 MinorVersion3)
```

Comment:
Change the implementation for checking the firmware. Do not hardcode the value inside the API. rather use a macro for checking a particular version @akhileshmalviya